### ISSUE TITLE
Locale Selector:  develop a service that determines the users locale based on ipcinfo or Geo-ip

### ISSUE THREAD

**Issue by jeffchew** [carbon-design-system/ibm-dotcom-library#515]
_hahnrob created the following on Oct 07:_  

 

#### User Story

> As a `[user role below]`:
As a developer of the Digital Design System Locale Selector

> I need to:
develop a service that uses the stored ipcinfo or the stored geo-ip info plus browser language preference

> so that I can:
enable the Locale Selector in the footer to accurately navigate global users to the correct local web pages

#### Additional information

- This issue is dependent on the design and the [package architecture https://github.com/carbon-design-system/ibm-dotcom-library/issues/515](https://zenhub.ibm.com/app/workspaces/digital-design-system-5b3a815a8bf6a932efa6fd03/issues/webstandards/digital-design/860) 
- [final design and specs](https://ibm.box.com/s/isie20a89socbo9ael687u6pb75uz9f8)
              - Includes:  Functional labels, Visual redlines, Behavior and interaction specs, Desktop and mobile
 - [Box link to Mary O'Donnell's functional specs](https://ibm.ent.box.com/folder/88263129224)
 - This issue is related to [issue 1815](https://zenhub.ibm.com/app/workspaces/digital-design-system-5b3a815a8bf6a932efa6fd03/issues/webstandards/digital-design/1815) and [Issue 1816](https://zenhub.ibm.com/app/workspaces/digital-design-system-5b3a815a8bf6a932efa6fd03/issues/webstandards/digital-design/1816)
 - This service will first fetch the ipcinfo cookie, if it exists, if not it will fetch the location and language based on geoip and browser language.  The output is an object containing cc-lc.
 - [US-EN country list](https://1.www.s81c.com/common/js/dynamicnav/www/countrylist/usen-utf8.js)
 - [France-French country list](https://1.www.s81c.com/common/js/dynamicnav/www/countrylist/frfr-utf8.js)

#### Acceptance criteria
 - [ ] Component has been tested using Netlify
 - [ ] The component development follows a modular component pattern
 - [ ] The component should be accessible via ES6 import
 - [ ] The component should be visible within the living documentation of the Dotcom components
 - [ ] Presented and approved Product Owner

 

_Original issue: https://github.ibm.com/webstandards/digital-design/issues/1817_

**Comment by jeffchew** [comment]
_On Oct 10, Anna-Wen commented:_
@Jeff-Chew @james-dow @Kenny-Lam Would this service also handle verification of cc-lc? Maybe we would want to create another util that would do that and include it in this service here.

AFTER DISCUSSION: 
Will include the verification logic in this service, the geo-ip util will only be responsible for returning the cc

**Comment by jeffchew** [comment]
@annawen1 I created and cleaned up all of the JSON files, attached here. I tried uploading though and my username specifically doesn't have access, I didn't try the functional ID but I'm assuming it may be the same thing. Going to reach out to the webmaster team about it.

[https://app.zenhub.com/files/193129061/315a5b69-5b05-4490-a50f-9f52e81e2328/download](https://app.zenhub.com/files/193129061/315a5b69-5b05-4490-a50f-9f52e81e2328/download)
